src/map/path.c

```diff
@@ -98,11 +99,9 @@ static int path_blownpos(struct block_list *bl, int16 m, int16 x0, int16 y0, int
 		ShowWarning("path_blownpos: count too many %d !\n",count);
 		count=25;
 	}
-	if( dx > 1 || dx < -1 || dy > 1 || dy < -1 ){
-		ShowError("path_blownpos: illegal dx=%d or dy=%d !\n",dx,dy);
-		dx=(dx>0)?1:((dx<0)?-1:0);
-		dy=(dy>0)?1:((dy<0)?-1:0);
-	}
+	Assert_retr((x0 << 16) | y0, dir >= UNIT_DIR_FIRST && dir < UNIT_DIR_MAX);
```

Comment:
Not true. If you'd have dx == 0 && dy == 0 it would still return the same, if you choose to use values of |dx| > 1 and |dy| > 1 you can't, it's ensured by using the enum unit_dir that you use a valid direction. The only existing call of path->blownpos still behaves the same, any custom code will need to obey, there's no advantage in using non-standard dx and dy values.